service/service.go

```diff
@@ -100,35 +14,103 @@ func (s *Service) GetDependency(dependencyKey string) (*Dependency, error) {
 	return nil, fmt.Errorf("dependency %s do not exist", dependencyKey)
 }
 
-// ValidateConfigurationEnv checks presence of env variables in mesg.yml under env section.
-func (s *Service) ValidateConfigurationEnv(env map[string]string) error {
-	var nonDefined []string
-	for key := range env {
-		exists := false
-		// check if "key=" exists in configuration
-		for _, env := range s.Configuration.Env {
-			if strings.HasPrefix(env, key+"=") {
-				exists = true
-			}
+// GetTask returns task taskKey of service.
+func (s *Service) GetTask(taskKey string) (*Task, error) {
+	for _, task := range s.Tasks {
+		if task.Key == taskKey {
+			return task, nil
 		}
-		if !exists {
-			nonDefined = append(nonDefined, key)
+	}
+	return nil, &TaskNotFoundError{
+		TaskKey:     taskKey,
+		ServiceName: s.Name,
+	}
+}
+
+// GetEvent returns event eventKey of service.
+func (s *Service) GetEvent(eventKey string) (*Event, error) {
+	for _, event := range s.Events {
+		if event.Key == eventKey {
+			return event, nil
 		}
 	}
-	if len(nonDefined) > 0 {
-		sort.Strings(nonDefined)
-		return ErrNotDefinedEnv{nonDefined}
+	return nil, &EventNotFoundError{
+		EventKey:    eventKey,
+		ServiceName: s.Name,
+	}
+}
+
+// ValidateTaskInputs produces warnings for task inputs that doesn't satisfy their parameter schemas.
+func (s *Service) ValidateTaskInputs(taskKey string, taskInputs map[string]interface{}) ([]*ParameterWarning, error) {
```

Comment:
```suggestion
func (s *Service) ValidateTaskInputs(taskKey string, taskInputs map[string]interface{}) error {
```

Left only error (the same for ValidateTaskOutputs). In general ParametrWarnings should be removed. If doing so, the `RequireTask*` won't be needed anymore.

Reply:
I agree but let's do it in another PR ;)

Reply:
https://github.com/mesg-foundation/engine/issues/1187